### PR TITLE
feat: add esm build

### PR DESCRIPTION
**Pull request by codeStryke**
I've attempted to create an `esm` build, but unfortunately both `marked` and `linkify-it` need to be included in the build because of the current interface.

I'm looking for ideas and suggestions how to make this better.

### CODE REVIEW

**Comment by UziTech** [comment]
Originally I just [copied](https://github.com/UziTech/marked-linkify-it/blob/7ce3866b40358c7a5361ec7adb6ae3fb8a03503a/src/helpers.js) the `inlineText` tokenizer to use it but I changed it to relying on marked. Maybe we just need to go back to copying it so we don't need to bundle marked.

**Comment by codeStryke** [comment]
I liked your approach since it would mean that any changes to that function would not have to be made twice but I don't see another way either. I've updated the PR, please take a look.

File: package.json [hunk 1]
```diff
@@ -10,7 +10,10 @@
   "scripts": {
     "test": "jest --verbose",
     "test:cover": "jest --coverage",
-    "lint": "eslint ."
+    "lint": "eslint .",
+    "build": "npm run rollup",
```
Comment (codeStryke) [inline package.json:14]: I copied these from [marked](https://github.com/markedjs/marked/blob/master/package.json#L77)

File: package.json [hunk 2]
```diff
@@ -10,7 +10,10 @@
...
+    "rollup": "npm run rollup:esm",
+    "rollup:esm": "rollup -c rollup.config.esm.js"
```
Comment (UziTech) [inline package.json:16]: I think we can simplify this.

```suggestion
    "build": "rollup -c rollup.config.esm.js"
```

**Review by UziTech — CHANGES_REQUESTED** [review]
Otherwise LGTM. I'll test it out a little more tonight.

File: rollup.config.esm.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { nodeResolve } from '@rollup/plugin-node-resolve';
```
Comment (codeStryke) [inline rollup.config.esm.js:1]: do these need to be converted to `require()` syntax?

**Review by UziTech — APPROVED** [review]

**Comment by codeStryke** [comment]
thank you! 🙏

**Pull request merged**